src/item_factory.cpp

```diff
@@ -257,12 +257,13 @@ void Item_factory::finalize_item_blacklist()
         for( auto &g : m_template_groups ) {
             g.second->remove_item( e.first );
         }
+
         recipe_dict.delete_if( [&]( recipe &r ) {
-            return r.result == e.first || r.requirements.remove_item( e.first );
+            return r.result == e.first || const_cast<requirement_data &>( r.requirements.obj() ).remove_item( e.first );
         } );
 
         remove_construction_if([&](construction &c) {
-            return c.requirements.remove_item( e.first );
+            return const_cast<requirement_data &>( c.requirements.obj() ).remove_item( e.first );
```

Comment:
You may have to change `requirement_data::remove_item` as well. Currently it returns `true` only if the item has actually been removed from the recipe (and the list is now empty).

If a requirement is shared by several recipes, it will be removed when the first recipe is checked here, and will *not* be removed when the second recipe (using the very same requirement object) is checked. `remove_item` will return false for the second recipe and it will be kept, even through its requirement is now invalid (has an empty list of items).